### ISSUE TITLE
JVM crashes with access violation during compile

### ISSUE THREAD

**Issue by RGriffin-Quest** [adoptium/adoptium-support#179]
## Summary

The JVM crashes frequently with an access violation. 

```
 A fatal error has been detected by the Java Runtime Environment:

  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007ffa4d5f783d, pid=18688, tid=20340

 JRE version: OpenJDK Runtime Environment AdoptOpenJDK (11.0.8+10) (build 11.0.8+10)
 Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK (11.0.8+10, mixed mode, tiered, compressed oops, g1 gc, windows-amd64)
 Problematic frame:
 V  [jvm.dll+0x16783d]

 No core dump will be written. Minidumps are not enabled by default on client versions of Windows

 An error report file with more information is saved as:
 C:\ .... \hs_err_pid18688.log

 Compiler replay data is saved as:
 C:\ ....\replay_pid18688.log

 If you would like to submit a bug report, please visit:
   https://github.com/AdoptOpenJDK/openjdk-support/issues
```

## Steps to reproduce

I don't have any easy steps to reproduce this but the compiler replay data is attached. If the compiler replay data is not enough then I might be able to come up with a small demo app.

## Expected results

Not crash!

## Actual results

Crash! 

## Triaging info

Java version:

openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)

What is your operating system and platform?

Microsoft Windows 10 Enterprise  10.0.18363 N/A Build 18363

How did you install Java?

From an MSI.

Did it work before?

We have just upgraded to a new version of the SecureBlackBox library from /n software and it appears that this library is causing the crash because the compiler replay data always shows compile secureblackbox/Base/SBZlib decompressEx as the last entry. The application ran for 28 minutes before crashing so the library would have been accessed well before the crash. I believe that the library is obfuscated because it contains packages and classes with names like a and b etc.  

Did you test with other Java versions?

Yes. Same result. 

openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)

If available, test with an upstream build:

Haven't tried this yet.

[replay_pid18688.log](https://github.com/AdoptOpenJDK/openjdk-support/files/5192012/replay_pid18688.log)
[hs_err_pid18688.log](https://github.com/AdoptOpenJDK/openjdk-support/files/5192014/hs_err_pid18688.log)

**Comment by aahlenst** [comment]
So that I get it right: You're compiling (as in `javac`) SecureBlackBox and while doing that, you get the access violation. Is that correct?

I had a look and the problem seems indeed to be in the JDK. In that case, we need a reproducer and verification with https://adoptopenjdk.net/upstream.html.

**Comment by RGriffin-Quest** [comment]
No this is while executing the application. This is HotSpot compiling the byte code. That is why is takes a while for the application to crash. 

Anyway I have received an unobfuscated version of the library from the vendor and will try that.

**Comment by RGriffin-Quest** [comment]
I'm working on creating a demo but it only crashes if run from within Eclipse. That means the command line arguments -agentlib:jdwp=transport=dt_socket,suspend=y,address=localhost:59876 which Eclipse adds must be causing the problem somehow. I will add more details tomorrow.

**Comment by RGriffin-Quest** [comment]
I have created a demo application that reproduces the bug. It only crashes when run from within Eclipse so I have put the source code in the attached zip file. It has one dependency on the included deploy.jar file. The application reads data from a Linux or UNIX host using SSH. The main class is com.quest.adk.demo.Demo. It takes 5 command line arguments:  

host port user password key

where host is the host name
port is the SSH port
user is the user name to login with
password is the password to login with
key is the key that unlocks the SecureBlackbox library.

Unfortunately I cannot supply the key I have because that is for use by our company only. However /n software who supply the library will send you a key to test with. Email support@nsoftware.com quoting support case number NS-SS071902158R

I have also tried a nightly build, 11.0.9+9-202009252343, and that failed too.

[demo.zip](https://github.com/AdoptOpenJDK/openjdk-support/files/5290031/demo.zip)
